src/java/arjdbc/jdbc/RubyJdbcConnection.java

```diff
@@ -1756,43 +2240,57 @@ protected IRubyObject integerToRuby(final ThreadContext context,
         final Ruby runtime, final ResultSet resultSet, final int column)
         throws SQLException {
         final long value = resultSet.getLong(column);
-        if ( value == 0 && resultSet.wasNull() ) return runtime.getNil();
+        if ( value == 0 && resultSet.wasNull() ) return context.nil;
         return runtime.newFixnum(value);
     }
 
     protected IRubyObject doubleToRuby(final ThreadContext context,
         final Ruby runtime, final ResultSet resultSet, final int column)
         throws SQLException {
         final double value = resultSet.getDouble(column);
-        if ( value == 0 && resultSet.wasNull() ) return runtime.getNil();
+        if ( value == 0 && resultSet.wasNull() ) return context.nil;
         return runtime.newFloat(value);
     }
 
+    private /* protected */ static final boolean useBytesForString = false; // TODO to be continued ...
+
     protected IRubyObject stringToRuby(final ThreadContext context,
         final Ruby runtime, final ResultSet resultSet, final int column) throws SQLException {
-        final String value = resultSet.getString(column);
-        if ( value == null && resultSet.wasNull() ) return runtime.getNil();
-        return RubyString.newInternalFromJavaExternal(runtime, value);
+        if ( useBytesForString ) { // optimized String -> byte[]
+            return bytesToRubyString(context, runtime, resultSet, column);
+        }
+        else {
+            final String value = resultSet.getString(column);
+            if ( value == null /* || resultSet.wasNull() */ ) return context.nil;
+            return RubyString.newInternalFromJavaExternal(runtime, value);
+        }
+    }
+
+    protected static IRubyObject bytesToRubyString(final ThreadContext context,
+        final Ruby runtime, final ResultSet resultSet, final int column)
+        throws SQLException { // optimized String -> byte[]
+        final byte[] value = resultSet.getBytes(column);
+        if ( value == null /* || resultSet.wasNull() */ ) return context.nil;
```

Comment:
I see this at least a couple of times.  Are you unsure about whether that || should be in there or not?  Either way I would say remove them and we can reinstall later if we need to.

Reply:
some drivers were fishy about the null checks - and since this is a base class for all ...
the original problem was that some of the wasNull check was rather expensive - unecessary
but than I am not sure at this point which/where ... the optimized string -> byte[] will be redone, I can remove this completely till than. was mostly a reference point to know it has been ported over.